backend/internal/db/dbgorm/triggers.go

```diff
@@ -0,0 +1,105 @@
+package dbgorm
+
+import (
+	"context"
+	"fmt"
+
+	"gorm.io/gorm/clause"
+
+	"go.autokitteh.dev/autokitteh/backend/internal/db/dbgorm/scheme"
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+func (db *gormdb) triggerToRecord(ctx context.Context, trigger sdktypes.Trigger) (*scheme.Trigger, error) {
+	connID := sdktypes.GetTriggerConnectionID(trigger)
+
+	conn, err := db.GetConnection(ctx, connID)
+	if err != nil {
+		return nil, fmt.Errorf("get trigger connection: %w", err)
+	}
+
+	projID := sdktypes.GetConnectionProjectID(conn)
+
+	envID := sdktypes.GetTriggerEnvID(trigger)
+	if envID != nil {
+		env, err := db.GetEnvByID(ctx, envID)
+		if err != nil {
+			return nil, fmt.Errorf("get trigger env: %w", err)
+		}
+
+		if projID.String() != sdktypes.GetEnvProjectID(env).String() {
+			return nil, fmt.Errorf("env and connection project mismatch: %s != %s", projID.String(), sdktypes.GetEnvProjectID(env).String())
+		}
+	}
+
+	return &scheme.Trigger{
+		TriggerID:    sdktypes.GetTriggerID(trigger).String(),
+		EnvID:        envID.String(),
+		ProjectID:    projID.String(),
+		ConnectionID: connID.String(),
+		EventType:    sdktypes.GetTriggerEventType(trigger),
+		CodeLocation: sdktypes.GetCodeLocationCanonicalString(sdktypes.GetTriggerCodeLocation(trigger)),
+	}, nil
+}
+
+func (db *gormdb) CreateTrigger(ctx context.Context, trigger sdktypes.Trigger) error {
+	t, err := db.triggerToRecord(ctx, trigger)
+	if err != nil {
+		return err
+	}
+
+	// Assuming if eventID already exists, nothing will happen and no error
+	if err := db.db.WithContext(ctx).Clauses(clause.OnConflict{DoNothing: true}).Create(t).Error; err != nil {
+		return translateError(err)
+	}
+	return nil
+}
+
+func (db *gormdb) UpdateTrigger(ctx context.Context, trigger sdktypes.Trigger) error {
+	t, err := db.triggerToRecord(ctx, trigger)
+	if err != nil {
+		return err
+	}
+
+	if err := db.db.WithContext(ctx).Updates(t).Error; err != nil {
+		return translateError(err)
+	}
+	return nil
+}
+
+func (db *gormdb) GetTrigger(ctx context.Context, mid sdktypes.TriggerID) (sdktypes.Trigger, error) {
+	return get(db.db, ctx, scheme.ParseTrigger, "trigger_id = ?", mid.String())
+}
+
+func (db *gormdb) DeleteTrigger(ctx context.Context, mid sdktypes.TriggerID) error {
+	var m scheme.Trigger
+	if err := db.db.WithContext(ctx).Where("trigger_id = ?", mid.String()).Delete(&m).Error; err != nil {
+		return translateError(err)
+	}
+
+	return nil
+}
```

Comment:
```suggestion
func (db *gormdb) DeleteTrigger(ctx context.Context, id sdktypes.TriggerID) error {
	var m scheme.Trigger
	if err := db.db.WithContext(ctx).Where("trigger_id = ?", id.String()).Delete(&m).Error; err != nil {
		return translateError(err)
	}

	return nil
}
```

do we want to handle not found somehow? currently I'm not sure if delete returns an error or not if record is not found

Reply:
i think it will fail silently, i am ok with it.